pkg/awsclient/elb_helper.go

```diff
@@ -0,0 +1,221 @@
+	instances := make([]*elb.Instance, 0)
+	for _, instance := range instanceIds {
+		instances = append(instances, &elb.Instance{InstanceId: aws.String(instance)})
+	}
+	i := &elb.DeregisterInstancesFromLoadBalancerInput{
+		Instances:        instances,
+		LoadBalancerName: aws.String(elbName),
+	}
+	_, err := c.DeregisterInstancesFromLoadBalancer(i)
+	return err
+}
+
+// DoesELBExist checks for the existence of an ELB by name. If there's an AWS
+// error it is returned.
+func (c *awsClient) DoesELBExist(elbName string) (bool, *AWSLoadBalancer, error) {
+	i := &elb.DescribeLoadBalancersInput{
+		LoadBalancerNames: []*string{aws.String(elbName)},
+	}
+	res, err := c.DescribeLoadBalancers(i)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case elb.ErrCodeAccessPointNotFoundException:
+				return false, &AWSLoadBalancer{}, nil
+			default:
+				return false, &AWSLoadBalancer{}, err
+			}
+		}
+	}
+	return true, &AWSLoadBalancer{ELBName: elbName, DNSName: *res.LoadBalancerDescriptions[0].DNSName, DNSZoneId: *res.LoadBalancerDescriptions[0].CanonicalHostedZoneNameID}, nil
+}
+
+// LoadBalancerInfo fields needed from NLBs
+type LoadBalancerInfo struct {
+	ExtCanonicalHostedZoneNameID string
+	ExtDNSName                   string
+	ExtLoadBalancerArn           string
+	ExtLoadBalancerName          string
+	ExtScheme                    string
+	IntCanonicalHostedZoneNameID string
+	IntDNSName                   string
+	IntLoadBalancerArn           string
+	IntLoadBalancerName          string
+	IntScheme                    string
+}
+
+// ListAllNLBs uses the DescribeLoadBalancersV2 to get back a list of all Network Load Balancers
+func (c *awsClient) ListAllNLBs() (LoadBalancerInfo, error) {
+	var LBInfo LoadBalancerInfo
+	i := &elbv2.DescribeLoadBalancersInput{}
+	output, err := c.DescribeLoadBalancersV2(i)
+	if err != nil {
+		return LoadBalancerInfo{}, err
+	}
+	for _, loadBalancer := range output.LoadBalancers {
+		scheme := loadBalancer.Scheme
+		if aws.StringValue(scheme) == "internet-facing" {
+			LBInfo.ExtCanonicalHostedZoneNameID = aws.StringValue(loadBalancer.CanonicalHostedZoneId)
+			LBInfo.ExtDNSName = aws.StringValue(loadBalancer.DNSName)
+			LBInfo.ExtLoadBalancerArn = aws.StringValue(loadBalancer.LoadBalancerArn)
+			LBInfo.ExtLoadBalancerName = aws.StringValue(loadBalancer.LoadBalancerName)
+		}
+		if aws.StringValue(scheme) == "internal" {
+			LBInfo.IntCanonicalHostedZoneNameID = aws.StringValue(loadBalancer.CanonicalHostedZoneId)
+			LBInfo.IntDNSName = aws.StringValue(loadBalancer.DNSName)
+			LBInfo.IntLoadBalancerArn = aws.StringValue(loadBalancer.LoadBalancerArn)
+			LBInfo.IntLoadBalancerName = aws.StringValue(loadBalancer.LoadBalancerName)
+		}
+	}
+	return LBInfo, nil
+}
+
+// DeleteExternalLoadBalancer takes in the external LB arn and deletes the entire LB
+func (c *awsClient) DeleteExternalLoadBalancer(extLoadBalancerArn string) error {
```

Comment:
For people wondering what happens if the load balancer is already deleted:

> If the load balancer does not exist or has already been deleted, the call succeeds.

[source](https://docs.aws.amazon.com/sdk-for-go/api/service/elbv2/#ELBV2.DeleteLoadBalancer)